### PR TITLE
EC2Architect fixed warning when domain is in use

### PR DESCRIPTION
**Pull request by Yash621**
This PR is in respect to issue #614 
@JackUrb please review it , I making sure that  url_safe_string method gets called, then checking the servers folder for files starting with "subdomain".json.

### CODE REVIEW

File: mephisto/abstractions/architects/ec2/ec2_architect.py [hunk 1]
```diff
@@ -156,6 +158,17 @@ def download_file(self, target_filename: str, save_dir: str) -> None:
                 if chunk:
                     out_file.write(chunk)
 
+    def check_domain_exists(self, subdomain_name: str):
+        """
+          Checks if the domain name exists in Route53
```
Comment (JackUrb) [inline mephisto/abstractions/architects/ec2/ec2_architect.py:163]: Docstring can be more direct on what's happening within:
```
"""
Checks to see if we have an active local record for the given subdomain
"""
```

File: mephisto/abstractions/architects/ec2/ec2_architect.py [hunk 2]
```diff
@@ -156,6 +158,17 @@ def download_file(self, target_filename: str, save_dir: str) -> None:
...
+        """
+        with open(self.server_detail_path, "r") as server_detail_file:
+            server_detail_file_files = server_detail_file.read().split('/n')
+            for file in server_detail_file_files:
+                if file.startswith(f'{subdomain_name}.json'):
+                    return True
+        return False
```
Comment (JackUrb) [inline mephisto/abstractions/architects/ec2/ec2_architect.py:170]: The check itself is really much simpler - we just need to see if the file at `self.server_detail_path` exists or not. If not, we're clear to create. If so, then there's potentially an issue.

File: mephisto/abstractions/architects/ec2/ec2_architect.py [hunk 3]
```diff
@@ -166,8 +179,13 @@ def assert_task_args(cls, args: DictConfig, shared_state: "SharedTaskState"):
         assert ec2_helpers.check_aws_credentials(
             profile_name
         ), "Given profile doesn't have registered credentials"
+        #   Producing a domain string that is safe for use
+        #   in ec2 resources
+        subdomain = url_safe_string(args.architect.subdomain)
+
+        assert cls.check_domain_exists(
```
Comment (JackUrb) [inline mephisto/abstractions/architects/ec2/ec2_architect.py:186]: We'd actually like to be sure that the domain _doesn't_ exist yet.

**Review by JackUrb — CHANGES_REQUESTED** [review]
Thanks for taking this one on! Some simple changes inline.

Also, more generally, I'm curious as to why you're making additional changes outside of the core files relevant to the feature, like those in `ec2_helpers.py`. These would be reverted by `black`, and are likely the cause of your branch not passing the style check at the moment.

**Comment by Yash621** [comment]
@JackUrb i have done all the changes that you suggested 
please review it !

**Pull request closed without merging**